tests/functional/aws-node-sdk/test/object/deleteMpu.js

```diff
@@ -38,59 +58,66 @@ describe('DELETE multipart', () => {
+                    process.stdout.write('Emptying bucket\n');
+                    return bucketUtil.empty(bucket)
+                    .then(() => {
+                        process.stdout.write('Deleting bucket\n');
+                        return bucketUtil.deleteOne(bucket);
+                    })
+                    .catch(err => {
+                        process.stdout.write('Error in afterEach');
+                        throw err;
+                    });
                 });
-            });
 
-            // AWS returns 404 - NoSuchUpload in us-east-1. This behavior can
-            // be toggled to be compatible with AWS by enabling usEastBehavior
-            // in the config.
-            itSkipIfAWS('should return 204 if mpu does not exist with uploadId',
-            done => {
-                const uploadId = 'nonexistinguploadid';
-                _assert204StatusCode(uploadId, done);
-            });
+                itSkipIfAWS('should return ${confLocation.name} if mpu does ' +
+                'not exist with uploadId',
+                done => {
+                    const uploadId = 'nonexistinguploadid';
+                    _assertStatusCode(uploadId, confLocation.statusCode, done);
+                });
 
-            describe('if mpu exists with uploadId + at least one part', () => {
-                let uploadId;
+                describe('if mpu exists with uploadId + at least one part',
+                () => {
+                    let uploadId;
 
-                beforeEach(() =>
-                    s3.createMultipartUploadAsync({
-                        Bucket: bucket,
-                        Key: key,
-                    })
-                    .then(res => {
-                        uploadId = res.UploadId;
-                        return s3.uploadPart({
+                    beforeEach(() =>
+                        s3.createMultipartUploadAsync({
                             Bucket: bucket,
                             Key: key,
-                            PartNumber: 1,
-                            UploadId: uploadId,
-                        });
-                    })
-                );
+                        })
+                        .then(res => {
+                            uploadId = res.UploadId;
+                            return s3.uploadPart({
+                                Bucket: bucket,
+                                Key: key,
+                                PartNumber: 1,
+                                UploadId: uploadId,
+                            });
+                        })
+                    );
 
-                it('should return 204 for abortMultipartUpload', done => {
-                    _assert204StatusCode(uploadId, done);
+                    it(`should return ${confLocation.statusCode} ` +
+                      'for abortMultipartUpload', done => {
+                        _assertStatusCode(uploadId, confLocation.statusCode,
```

Comment:
shouldn't it return 204 in all cases? (if `uploadId` exists, even in legacy behavior should not return 404)
unless I am misunderstanding what you are asserting?